### PR TITLE
Protobuf definition for autonaming configuration

### PR DESCRIPTION
**Pull request by mikhailshilkov**
Introduces protobuf changes required for [Autonaming](https://github.com/pulumi/pulumi/discussions/17592) implementation.

We add a struct field to CheckRequest. The struct contains two fields:
- Proposed name that will hold a name calculated by the engine which is a suggested candidate for resource's autoname (in case it's needed)
- Mode of how to apply that name:
  - Disable: there should be no proposed name passed, since the autonaming is disabled for this resource
  - Enforce: provider _must_ use the proposed name if no explicit name was provider, or error otherwise
  - Propose: use the provided name as a baseline but can adjust it given provider's domain knowledge

Once this ships, we can start upgrading providers to this protocol version and implementing autonaming provider-side.

Also, we add a `SupportsAutonamingConfiguration` flag that a provider needs to set to `true` in `ConfigureResponse` once it declares that it will respec autonaming configuration provided by the engine.

Part of https://github.com/pulumi/pulumi/issues/1518

### CODE REVIEW

File: proto/pulumi/provider.proto [hunk 1]
```diff
@@ -456,6 +456,27 @@ message CheckRequest {
     // The type of the resource being checked. This must match the type specified by the `urn` field, and is passed so
     // that providers do not have to implement URN parsing in order to extract the type of the resource.
     string type = 7;
+
+    // Configuration for automatic resource naming behavior. This structure contains fields that control how the provider
+    // handles resource names, including proposed names and naming modes.
+    message AutonamingOptions {
+        // The proposed name for the resource being checked. This may be used by the provider as a suggestion
+        // for the final resource name, depending on the specified mode.
+        string proposedName = 1;
```
Comment (Frassle) [inline proto/pulumi/provider.proto:465]:
```suggestion
        string proposed_name = 1;
```
proto files should use snake case for properties https://protobuf.dev/programming-guides/style/#message-field-names

Comment (mikhailshilkov) [reply proto/pulumi/provider.proto:465]: What about neighbour `randomSeed` and `secretNumber` a few lines above?

Comment (Frassle) [reply proto/pulumi/provider.proto]: they're old, and were added before we started consistently using this rule. I don't know if it's safe to rename them.

Comment (mikhailshilkov) [reply proto/pulumi/provider.proto]: Anyway, I updated to underscore

**Pull request merged**